### DISCUSSION TITLE
Dashboard login is no longer working

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/aspire#4826] [Q&A]
Hello,

This is my first time interacting with development teams on GitHub, I hope I am not doing it wrong.

I started build an aspire app and deployed it to Azure a month ago. After deployment, I could access to Dashboard and see deployed services, logs, traces...

After a while and a couple of deployments, the dashboard login does no longer works, I get the error: "You can't sign in here with a personal account. Use your work or school account instead."

I tried creating a fresh solution out of the box and deploying directly to azure without any code change, I get the same error.

I know this might an Azure issue but I am curious about why this worked for a while and then stopped? Thanks in advance for your help!

EDIT: I am not using a standalone dashboard, this issue can be reproduced using the ".Net Aspire Starter Application" project template.

**Comment by davidebbo** [comment]
First time trying to deploy an Aspire solution to ACA, and I get the same issue.

**Reply by ghost** [reply]
yes, this might be an AZD issue more than an Aspire issue. you have to add yourself as an owner in Azure Entre ID registered apps:
![image](https://github.com/user-attachments/assets/e75e93e2-f7bd-417d-9b82-9c96e4715a04)

**Comment by davidebbo** [comment]
Interesting. How do you get to this application? If I go to https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade, I only see old irrelevant apps under both 'All applications', and 'Applications from personal account'. Did the one you're looking at get created as part of azd deployment?

**Reply by ghost** [reply]
You should go to Entra Id => App registration => Look for the app your Aspire deployment created (for me it starts with "az-dev-")
I added myself as owner in the latest one and it worked for all subsequent deployments...

Hope it helps!

**Reply by davidebbo** [reply]
Thanks @samimejri. That's actually the same place where I was. But for some reason, there is no app there that was created by the deployment. I just see a couple super old test apps that I know are not related.

In any case, it feels like a bug if using this requires people to manually add themselves as an owner. I agree it's more of an azd/Azure bug, but may as well keep it here for now since it directly affects the Aspire experience.

**Reply by ghost** [reply]
Are deploying using azd on a terminal or via Github Actions? I think the new app registration are created when you use Github Actions...

**Reply by davidebbo** [reply]
Oh, interesting, I'm doing it from my local machine. I'll have to play around with Github Actions. Though presumably it should be possible to create the new app registration manually. What I don't understand is: what associates the Dashboard with this specific App registration?

Other interesting thing: from the container app / Overview / .NET Aspire Dashboard settings, I unchecked ".NET Aspire Dashboard" and saved, just to see. The operation was successful, but then when I refreshed, it was still checked!

Seems a number of things related to running the Dashboard is ACA are kind of buggy / half cooked...

**Reply by ghost** [reply]
> Oh, interesting, I'm doing it from my local machine. I'll have to play around with Github Actions. Though presumably it should be possible to create the new app registration manually. What I don't understand is: what associates the Dashboard with this specific App registration?

It kind of defeats the purpose of Aspire and IaC! I don't think we should have that deep of a knowledge...

> Other interesting thing: from the container app / Overview / .NET Aspire Dashboard settings, I unchecked ".NET Aspire Dashboard" and saved, just to see. The operation was successful, but then when I refreshed, it was still checked!

Same thing for me. But we should blame Azure for that?

> Seems a number of things related to running the Dashboard is ACA are kind of buggy / half cooked...

That makes us early adopters :)

**Reply by davidebbo** [reply]
> That makes us early adopters :)

Right, I'm sure none of this is by design. They are just early quirks that will get fixed up.

Someone else mentioned running into this on Discord. See [thread](https://discord.com/channels/732297728826277939/759125320505884752/threads/1261733988209524818).

**Reply by davidebbo** [reply]
@samimejri see https://github.com/Azure/azure-dev/issues/4010 for more details on this issue.

**Reply by davidebbo** [reply]
The active issue is now https://github.com/Azure/azure-dev/issues/4121
